jaeger-core/src/main/java/com/uber/jaeger/propagation/B3TextMapCodec.java

```diff
@@ -47,44 +49,52 @@
+        carrier.put(PARENT_SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getParentId()));
+      }
+      carrier.put(SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getSpanId()));
+      carrier.put(SAMPLED_NAME, spanContext.isSampled() ? "1" : "0");
+      if (spanContext.isDebug()) {
+        carrier.put(FLAGS_NAME, "1");
+      }
+    } catch (RuntimeException ex) {
+      log.error("Error when injecting SpanContext into carrier. Handling gracefully.", ex);
     }
   }
 
   @Override
   public SpanContext extract(TextMap carrier) {
-    Long traceId = null;
-    Long spanId = null;
-    Long parentId = 0L; // Conventionally, parent id == 0 means the root span
-    byte flags = 0;
-    for (Map.Entry<String, String> entry : carrier) {
-      if (entry.getKey().equalsIgnoreCase(SAMPLED_NAME)) {
-        String value = entry.getValue();
-        if ("1".equals(value) || "true".equalsIgnoreCase(value)) {
-          flags |= SAMPLED_FLAG;
-        }
-      } else if (entry.getKey().equalsIgnoreCase(TRACE_ID_NAME)) {
-        traceId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
-      } else if (entry.getKey().equalsIgnoreCase(PARENT_SPAN_ID_NAME)) {
-        parentId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
-      } else if (entry.getKey().equalsIgnoreCase(SPAN_ID_NAME)) {
-        spanId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
-      } else if (entry.getKey().equalsIgnoreCase(FLAGS_NAME)) {
-        if (entry.getValue().equals("1")) {
-          flags |= DEBUG_FLAG;
+    try {
+      Long traceId = null;
+      Long spanId = null;
+      Long parentId = 0L; // Conventionally, parent id == 0 means the root span
+      byte flags = 0;
+      for (Map.Entry<String, String> entry : carrier) {
+        if (entry.getKey().equalsIgnoreCase(SAMPLED_NAME)) {
+          String value = entry.getValue();
+          if ("1".equals(value) || "true".equalsIgnoreCase(value)) {
+            flags |= SAMPLED_FLAG;
+          }
+        } else if (entry.getKey().equalsIgnoreCase(TRACE_ID_NAME)) {
+          traceId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
+        } else if (entry.getKey().equalsIgnoreCase(PARENT_SPAN_ID_NAME)) {
+          parentId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
+        } else if (entry.getKey().equalsIgnoreCase(SPAN_ID_NAME)) {
+          spanId = HexCodec.lowerHexToUnsignedLong(entry.getValue());
+        } else if (entry.getKey().equalsIgnoreCase(FLAGS_NAME)) {
+          if (entry.getValue().equals("1")) {
+            flags |= DEBUG_FLAG;
+          }
         }
       }
-    }
 
-    if (null != traceId && null != parentId && null != spanId) {
-      return new SpanContext(traceId, spanId, parentId, flags);
+      if (null != traceId && null != parentId && null != spanId) {
+        return new SpanContext(traceId, spanId, parentId, flags);
+      }
+    } catch (RuntimeException ex) {
```

Comment:
We might want to revert back https://github.com/jaegertracing/jaeger-client-java/pull/355 as it removed exceptions from  `lowerHexToUnsignedLong`

Reply:
I am not sure, I left my comment on that here: https://github.com/jaegertracing/jaeger-client-java/issues/360#issuecomment-369161607

Reply:
without throwing ex a corrupted context wouldn't be logged as we simply return null in that case.

Reply:
that's true, but I would rather add a simple log message somewhere as we know the cause instead of logging full stack trace.